clients/src/test/java/org/apache/kafka/clients/MockClient.java

```diff
@@ -135,6 +136,12 @@ public boolean connectionFailed(Node node) {
     }
 
     @Override
+    public AuthenticationException authenticationException(Node node) {
+        //TODO
```

Comment:
TODO can be removed since we are not using this in tests?

Reply:
Maybe we should add a JIRA to implement this? Seems like a testing gap if we are not needing it since most of the consumer unit tests depend on `MockClient`. We should have test cases to ensure the behavior if there is an authentication failure in any of the various APIs.

Reply:
Yes, we should have a JIRA to implement unit tests using the `MockClient`. The tests in this PR do actual authentication. @vahidhashemian Would you like to work on that in a separate PR?

Reply:
Opened another JIRA for mock tests: https://issues.apache.org/jira/browse/KAFKA-5944